src/index.js

```diff
@@ -98,10 +99,26 @@ async function monitor(options) {
         process.exit(1);
       }
     });
-    process.on('unhandledRejection', (reason, p) => {
+    process.on('unhandledRejection', async (reason, p) => {
       let err = 'Unhandled Rejection at: Promise ' + p + ' reason: ' + reason;
       console.log(err);
-      m.reportError(err, 'warning');
+      if (options.unhandledRejectionBail) {
```

Comment:
Please do:
```js
if (!options.unhandledRejectionBail) {
  m.reportError(err, 'warning');
  return;
}

....
```

So we return early... rather than having a `else`